SerialPrograms/Source/PokemonSV/Programs/ItemPrinter/PokemonSV_ItemPrinterRNG.cpp

```diff
@@ -156,12 +158,32 @@ ItemPrinterRNG::ItemPrinterRNG()
 //        3
 //    )
 //    , MATERIAL_FARMER_DISABLED_EXPLANATION("")
-    , MATERIAL_FARMER_JOBS_PERIOD(
+    , MATERIAL_FARMER_TRIGGER(
+        "<b>Trigger to start material farmer:</b><br>",
+        {
+            {MaterialFarmerTrigger::FIXED_NUM_PRINT_JOBS, "fixed", "Start Material Farmer when done a certain number of print jobs."},
+            {MaterialFarmerTrigger::MINIMUM_HAPPINY_DUST, "happiny-dust", "Start Material Farmer when Happiny Dust is less than a certain number."},
+        },
+        LockMode::UNLOCK_WHILE_RUNNING,
+        MaterialFarmerTrigger::FIXED_NUM_PRINT_JOBS
+    )
+    , MATERIAL_FARMER_FIXED_NUM_JOBS(
         "<b>Print Jobs per Material Farming Session:</b><br>"
         "Run the material farmer once for this many jobs printed.",
         LockMode::UNLOCK_WHILE_RUNNING, 500,
         20, 650
     )
+    , MIN_HAPPINY_DUST(
+        "<b>Minimum Happiny Dust:</b><br>"
+        "Run the material farmer before the number of Happiny Dust drops "
+        "below this number.<br>"
+        "This ensures no other material drops below this number. "
+        "If a material starts below this threshold, it remains there.<br>"
+        "Changes to this number only take place after returning to "
+        "the item printer, after material farming.",
+        LockMode::UNLOCK_WHILE_RUNNING, 100,

```

Comment:
I dunno if setting it as low as 100 is necessarily the # since the auto-select always uses the ones with the most quantity first. So if you have larger quantities of non-Happiny dust that are harder to collect, it will consume those as well.

But I'm not sure how much material is used for making TMs. Maybe 100 is enough for most purposes.

Reply:
TMs usually use 3 for a given material. Occasionally up to 8. 
https://bulbapedia.bulbagarden.net/wiki/TM_Material

I can change the default to 400.

If you run the material farmer for 30 minutes, you'll get around 450 Happiny Dust. So a threshold of around 400 is reasonable. If you run the material for 60 minutes, you'll get around 900 Happiny Dust, so a threshold of around 100 would be better in that case.